### ISSUE TITLE
SNOW-876360: Ensure Compatibility: Upgrade cryptography to Support Version 41.X.X

### ISSUE THREAD

**Issue by david-insait** [snowflakedb/snowflake-connector-python#1667]
### What is the current behavior?

When installing the latest version of the cryptography package for security purposes, I encountered the following ERROR related to a pip dependency:
```
ERROR: pip's dependency resolver does not currently consider all the installed packages, leading to the following dependency conflicts:
snowflake-connector-python 3.0.4 requires cryptography<41.0.0,>=3.1.0, but you have cryptography 41.0.2 which is incompatible.

```
This issue arises due to the incompatibility between the required version of cryptography (41.0.0 or lower) by snowflake-connector-python 3.0.4 and the currently installed version (41.0.2).

### What is the desired behavior?

Ensure Compatibility: Update snowflake-connector-python to Support the Latest Current Version of cryptography (41.0.2)

### How would this improve `snowflake-connector-python`?

This enhancement aims to improve the package for both security and compatibility reasons

### References and other background

https://security.snyk.io/package/pip/cryptography

**Comment by sfc-gh-achandrasekaran** [comment]
@david-insait we will be releasing a new connector next Monday with this change

**Comment by sfc-gh-dszmolka** [comment]
resolved in #1629
